databricks-sdk-java/src/main/java/com/databricks/sdk/core/ApiClient.java

```diff
@@ -63,11 +70,12 @@ public ApiClient(DatabricksConfig config, Timer timer) {
       debugTruncateBytes = 96;
     }
 
-    maxAttempts = 3;
+    maxAttempts = 4;
```

Comment:
Just curious: our other SDKs implement retries based on timeouts: retry until time has elapsed, then fail. Would it be problematic for your use case? Either way, let's not make that change in this PR to reduce complexity.

Reply:
I think that would be a good idea here as well i.e. having an overall timeout when we stop retrying